cumulusci/core/config/project_config.py

```diff
@@ -234,27 +236,51 @@ def _log_detected_overrides_as_warning(self, info):
     def _split_repo_url(self, url):
         url_parts = url.rstrip("/").split("/")
         name = url_parts[-1]
-        owner = url_parts[-2]
+        owner = url_parts[-2].split(":")[-1]
         if name.endswith(".git"):
             name = name[:-4]
         git_info = {"url": url, "owner": owner, "name": name}
         return git_info
 
+    def git_path(self, tail=None):
+        """Returns a Path to the .git directory in self.repo_root
+        with tail appended (if present) or None if self.repo_root
+        is not set."""
+        path = None
+        if self.repo_root:
+            path = Path(self.repo_root) / ".git"
+            if tail is not None:
+                path = path / str(tail)
+        return path
+
+    def git_config_remote_origin_line(self, start_string):
+        """Rerturns the first line under the [remote origin]
+        section of the .git/config file that that starts
+        with start_string. Returns None if .git/config file
+        not present or no matching line is found."""
+        with open(self.git_path("config"), "r") as f:
+            for line in dropwhile(self.line_not_remote_origin, f):
+                if start_string in line.strip():
+                    return line.strip()
```

Comment:
Hmm, how about

```suggestion
            while next(f).strip() != '[remote "origin"]':
                continue
            for line in f:
                if start_string in line.strip():
                    return line.strip()
```

Reply:
I don't mind your suggested version here, I thought I was being slick with the use of `dropwhile()`. Are you thinking that readability is better this way, or don't like the use of dropwhile() here? Just curious.

Reply:
I think mostly I didn't like having to look elsewhere to find self.line_not_remote_origin when it turned out to be a one-liner.